lib/features/cli/networking/download_binary.dart

```diff
@@ -0,0 +1,22 @@
+import 'dart:io';
+
+import '../../../core/async/event_synchronizer.dart';
+import '../../../core/networking/protocol/project_build.dart';
+
+/// Requests the binary of the last build of the [session] and writes it
+/// to the [outFile].
+Future<void> downloadBinary(EventSynchronizer session, File outFile) async {
+  // Request the produced binary file
+  await session.request(getBinaryRequest);
+  final bin = await session.expect(
+    BinaryResponse,
```

Comment:
Should we maybe rename `BinaryResponse` to `BinaryFileResponse` to avoid confusion?

Edit: In `download_logs.dart` I figured out that `BinaryResponse` actually just is a response in binary and not specific to the _executable_... The naming is a little confusing here.

Reply:
How else should a response containing binary data be named? What do u propose?

Reply:
How about `ResponseAsBinary` or `ResponseInBinary`? ^^